impl/src/main/java/org/jboss/weld/contexts/CreationalContextImpl.java

```diff
@@ -89,6 +96,7 @@ private CreationalContextImpl() {
         // We can't use immutable empty lists because of some rare scenarios with Instance.get()
         this.dependentInstances = Collections.synchronizedList(new ArrayList<ContextualInstance<?>>());
         this.parentDependentInstances = Collections.synchronizedList(new ArrayList<ContextualInstance<?>>());
+        this.destroyed = new HashSet<>();
```

Comment:
@ljnelson I think you might want to be lazy here and avoid the allocation if you can because many contexts won't need it. Start with `null` and, since you are only using this in a synchronized block, only create the `HashSet` when first needed.

Reply:
Done